File: src/test/java/com/github/liebharc/queryenrichment/H2PlanBuilder.java
package com.github.liebharc.queryenrichment;

import java.sql.Connection;
import java.sql.Statement;
import java.util.List;

public class H2PlanBuilder extends PlanBuilder {

    private final Connection connection;

    public H2PlanBuilder(Connection connection, List<Step<?>> steps) {
        super(steps);
        this.connection = connection;
    }

    @Override
    protected QueryBuilder getQueryBuilder() {
        return new H2QueryBuilder(connection);
    }
}

File: src/test/java/com/github/liebharc/queryenrichment/H2PlanBuilder.kt
package com.github.liebharc.queryenrichment

import java.sql.Connection

class H2PlanBuilder(private val connection: Connection?, steps: List<ExecutableStep<*, Any?>>) : PlanBuilder<Any?>(steps) {

    override val queryBuilder: QueryBuilder
        get() = H2QueryBuilder(connection!!)
}
